### PR TITLE
GBT probabilistic classification

### PR DESCRIPTION
**Pull request by ShvetsKS**

### CODE REVIEW

**Comment by ShvetsKS** [comment]
make test

File: algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i [hunk 1]
```diff
@@ -73,23 +75,92 @@ public:
...
-            res[iRow] = label[services::internal::SignBit<algorithmFPType, cpu>::get(res[iRow])];
+            WriteOnlyRows<algorithmFPType, cpu> resBD(this->_res, 0, 1);
+            DAAL_CHECK_BLOCK_STATUS(resBD);
+            const algorithmFPType label[2] = { algorithmFPType(1.), algorithmFPType(0.) };
+            algorithmFPType* res = resBD.get();
+            WriteOnlyRows<algorithmFPType, cpu> probBD(_prob, 0, 1);
+            DAAL_CHECK_BLOCK_STATUS(probBD);
+            algorithmFPType* prob_pred = probBD.get();
+            TArray<algorithmFPType, cpu> expValPtr(nRows);
+            algorithmFPType* expVal = expValPtr.get();
```
Comment (vepifanov) [inline algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i]: Add DAAL_CHECK_MALLOC here.

File: algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i [hunk 2]
```diff
@@ -73,23 +75,92 @@ public:
...
+        }
+
+        else if ((!this->_res) && _prob)
+        {
+            WriteOnlyRows<algorithmFPType, cpu> probBD(_prob, 0, 1);
+            DAAL_CHECK_BLOCK_STATUS(probBD);
+            algorithmFPType* prob_pred = probBD.get();
+            TArray<algorithmFPType, cpu> expValPtr(nRows);
+            algorithmFPType* expVal = expValPtr.get();
+            NumericTablePtr expNT = HomogenNumericTableCPU<algorithmFPType, cpu>::create(expVal, 1, nRows, &s);
```
Comment (vepifanov) [inline algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i]: Add DAAL_CHECK_MALLOC here.

File: algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i [hunk 3]
```diff
@@ -192,38 +265,87 @@ services::Status PredictMulticlassTask<algorithmFPType, cpu>::predictByAllTrees(
 {
     WriteOnlyRows<algorithmFPType, cpu> resBD(_res, 0, 1);
     DAAL_CHECK_BLOCK_STATUS(resBD);
-    const size_t nCols(_data->getNumberOfColumns());
-    ClassesRawBoostedTls lsData(nClasses*VECTOR_BLOCK_SIZE);
 
+    const size_t nCols(_data->getNumberOfColumns());
+    const size_t nRows(_data->getNumberOfRows());
+    //ClassesRawBoostedTls lsData(nClasses*VECTOR_BLOCK_SIZE);
```
Comment (vepifanov) [inline algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i]: Delete comment.

File: algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i (hunk 2 above)
Comment (ShvetsKS) [reply algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i]: done. Thanks

File: algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i (hunk 3 above)
Comment (ShvetsKS) [reply algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i]: done. Thanks

File: algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i (hunk 1 above)
Comment (ShvetsKS) [reply algorithms/kernel/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i]: dona. Thanks

**Comment by vmeshche** [comment]
make test

**Pull request merged**